exchanges/stream/websocket_test.go

```diff
@@ -92,416 +97,267 @@ type dodgyConnection struct {
-	}
+	assert.ErrorIs(t, err, errInvalidWebsocketURL, "Setup should error correctly")
 
 	websocketSetup.RunningURL = "wss://www.google.com"
 	websocketSetup.RunningURLAuth = "http://www.google.com"
 	err = w.Setup(websocketSetup)
-	if !errors.Is(err, errInvalidWebsocketURL) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, errInvalidWebsocketURL)
-	}
+	assert.ErrorIs(t, err, errInvalidWebsocketURL, "Setup should error correctly")
 
 	websocketSetup.RunningURLAuth = "wss://www.google.com"
 	err = w.Setup(websocketSetup)
-	if !errors.Is(err, errInvalidTrafficTimeout) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, errInvalidTrafficTimeout)
-	}
+	assert.ErrorIs(t, err, errInvalidTrafficTimeout, "Setup should error correctly")
 
 	websocketSetup.ExchangeConfig.WebsocketTrafficTimeout = time.Minute
 	err = w.Setup(websocketSetup)
-	if !errors.Is(err, nil) {
-		t.Fatalf("received: %v but expected: %v", err, nil)
-	}
+	assert.NoError(t, err, "Setup should not error")
 }
 
 func TestTrafficMonitorTimeout(t *testing.T) {
 	t.Parallel()
-	ws := *New()
-	if err := ws.Setup(defaultSetup); err != nil {
-		t.Fatal(err)
-	}
-	ws.trafficTimeout = time.Second * 2
+	ws := NewWebsocket()
+	err := ws.Setup(defaultSetup)
+	require.NoError(t, err, "Setup must not error")
+
+	ws.trafficTimeout = time.Millisecond * 42
 	ws.ShutdownC = make(chan struct{})
 	ws.trafficMonitor()
-	if !ws.IsTrafficMonitorRunning() {
-		t.Fatal("traffic monitor should be running")
-	}
+	assert.True(t, ws.IsTrafficMonitorRunning(), "traffic monitor should be running")
+
 	// Deploy traffic alert
 	ws.TrafficAlert <- struct{}{}
 	// try to add another traffic monitor
 	ws.trafficMonitor()
-	if !ws.IsTrafficMonitorRunning() {
-		t.Fatal("traffic monitor should be running")
-	}
+	assert.True(t, ws.IsTrafficMonitorRunning(), "traffic monitor should be running")
+
 	// prevent shutdown routine
-	ws.setConnectedStatus(false)
+	ws.setState(disconnected)
 	// await timeout closure
 	ws.Wg.Wait()
-	if ws.IsTrafficMonitorRunning() {
-		t.Error("should be dead")
-	}
+	assert.False(t, ws.IsTrafficMonitorRunning(), "traffic monitor should be not be running")
 }
 
 func TestIsDisconnectionError(t *testing.T) {
 	t.Parallel()
-	isADisconnectionError := IsDisconnectionError(errors.New("errorText"))
-	if isADisconnectionError {
-		t.Error("Its not")
-	}
-	isADisconnectionError = IsDisconnectionError(&websocket.CloseError{
-		Code: 1006,
-		Text: "errorText",
-	})
-	if !isADisconnectionError {
-		t.Error("It is")
-	}
-
-	isADisconnectionError = IsDisconnectionError(&net.OpError{
-		Err: errClosedConnection,
-	})
-	if isADisconnectionError {
-		t.Error("It's not")
-	}
-
-	isADisconnectionError = IsDisconnectionError(&net.OpError{
-		Err: errors.New("errText"),
-	})
-	if !isADisconnectionError {
-		t.Error("It is")
-	}
+	assert.False(t, IsDisconnectionError(errors.New("errorText")), "IsADisconnectionError should return false")
```

Comment:
```suggestion
	assert.False(t, IsDisconnectionError(errors.New("errorText")), "IsDisconnectionError should return false")
```

For this one and the blow

Reply:
This is so funny. I was copying the test messages and thinking "IsADisc... is a stupid name but I'm not going to fix it, I've done enough" ... so karma dictates I should learn from being lazy - it was wrong after all anyway!

Fixed 52ebf55a